helpers/network-recorder.js

```diff
@@ -21,14 +21,20 @@ const NetworkManager = require('./web-inspector').NetworkManager;
 const REQUEST_FINISHED = NetworkManager.EventTypes.RequestFinished;
 
 class NetworkRecorder {
-  constructor(recordArray) {
+  constructor(recordArray, requestListener) {
     this._records = recordArray;
 
+    this._listener = requestListener;
+
     this.networkManager = NetworkManager.createWithFakeTarget();
 
     // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
     this.networkManager.addEventListener(REQUEST_FINISHED, request => {
       this._records.push(request);
+
+      if (this._listener) {
```

Comment:
yeah, I was thinking the same thing, but that really goes hand-in-hand with managing the network recording lifecycle outside of a gather, so I wanted to do this the simplest way possible while we start to see how #81 takes form.